### PR TITLE
Port to Python 3

### PR DESCRIPTION
**Pull request by octamois**
- Open for testing
- Report bugs at anmolmishra.jiit@gmail.com

### CODE REVIEW

File: library/pippy/physics/myelements/elements.py [hunk 1]
```diff
@@ -410,7 +410,7 @@ def mouse_move(self, pos):
             self.mouseJoint.target = (x, y)
 
     def pickle_save(self, fn, additional_vars={}):
-        import cPickle as pickle
+        import pickle as pickle
```
Comment (quozl) [inline library/pippy/physics/myelements/elements.py]: Should this be `import pickle`?

File: pippy_app.py [hunk 2]
```diff
@@ -1432,8 +1432,8 @@ def main():
     if False:  # Change this to True to test within Pippy
         sys.argv = sys.argv + ['-d', '/tmp', 'Pippy',
                                '/home/olpc/pippy_app.py']
-    print(_('Working...'))
+    print((_('Working...')))
```
Comment (quozl) [inline pippy_app.py]: Are the extra parentheses required?  Also on line 1438, and line 1423.

**Comment by quozl** [comment]
Thanks.  Reviewed to 2d30432ae61f062cfb16d1da01fb930ddb344502.  Not tested.  @Pro-Panda, could you test?

**Comment by Pro-Panda** [comment]
Thanks.
Tested using [sugar-toolkit-gtk3/@4b294ac](https://github.com/Pro-Panda/sugar-toolkit-gtk3/tree/python3-port)
 - the activity still calls the `Python 2` toolkit code; it should call the `Python 3` version
 - Following errors were reported while running the activity:
```shell
1529724998.263839 ERROR root: Error saving activity object to datastore
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/sugar3/activity/activity.py", line 1257, in _prepare_close
    self.save()
  File "/usr/lib/python2.7/dist-packages/sugar3/activity/activity.py", line 958, in save
    self.write_file(file_path)
  File "/usr/share/sugar/activities/Pippy/pippy_app.py", line 1008, in write_file
    data = self._source_tabs.get_all_data()
  File "/usr/share/sugar/activities/Pippy/notebook.py", line 321, in get_all_data
    label = self._purify_file(self.get_tab_label(child).get_text())
  File "/usr/share/sugar/activities/Pippy/notebook.py", line 307, in _purify_file
    unicodedata.normalize('NFKD', label).encode('ascii', 'ignore')
TypeError: normalize() argument 2 must be unicode, not str
```
 - The extra paranthesis is due to a regex issue in 2to3 (which you'd be presumably using), a possible solution would be to review each change made before comitting; `git add -p` could help
 - Pippy uses `telepathy` which should be ported to `TelepathyGLib`; please see updated [guide](https://github.com/sugarlabs/sugar-docs/blob/master/src/python-porting-guide.md) to port activities to Python 3

**Comment by octamois** [comment]
@Pro-Panda Thanks, working!

**Comment by octamois** [comment]
@Pro-Panda , I missed to update activity.info so this is calling Python 2 toolkit code

**Comment by divyanshu-rawat** [comment]
@octamois 👍

**Comment by pro-panda** [comment]
Stale; Needs more work.

@octamois if you plan to fix the same, please reopen the PR with the fixes

**Pull request closed without merging**